core/che-core-api-model/src/main/java/org/eclipse/che/api/core/model/workspace/runtime/RuntimeIdentity.java

```diff
@@ -16,5 +16,7 @@
 
   String getEnvName();
 
-  String getOwner();
+  String getOwnerName();
+
+  String getOwnerId();
```

Comment:
Do we really need name and id at the same time? I would consider replacing username with user id. 
For me, it is better because:
1. We already have workspace id here but not workspace name (it's a weak reason but still it is better to do in the same way).
2. Username is potentially modifiable value, while user id is unmodifiable. So if username from `RuntimeIdentity` will be stored somewhere like DB, logs then it can be false.

@gazarenkov WDYT?

Reply:
I agree that having both `userId` and `userName` might seem redundant, but it seems that best way to make this PR non-breaking and avoid making it even bigger and more complicated. For example I assume that the owner was used to build the name of the created workspace in the dashboard, and in certainly in some other places, and I'm not sure we want to have this replaced with the `userId`.
Maybe using only the `userId`, and replace the use of the `ownerName` field by calculated / retrieved values at use sites might be done in a distinct PR, no ?

Reply:
@davidfestal I'd way that your changes are already a bit breaking because you rename the field.
I'm OK to replace name value with id in another PR but then let's do not add owner id in this PR.
If you need it in rh-che, you can retrieve user id by username from RuntimeIdentity using UserManager class. I think it can be used as a temporary solution.
WDYT?

Reply:
well, it seems to me it's a bit just going backwards. But if you all want me to do that, I can do it. @l0rd wdyt ?

Reply:
I think it's better to get agreement how it should look like and only then rework it. Anyway, your job can be saved in another branch and it could be used later.

Reply:
Sure. I'll wait for other thoughts then.

Reply:
What if we just add ```ownerId``` field?

Reply:
I mean left method ```String getOwner();``` as is and not rename it and also add method ```String getOwnerId();```

Reply:
I don't see what it would have simplified, and things would be less clear IMO.

Reply:
1. It's backward compatible. Nothing has been removed or renamed.
2.It's adding a method to get owner id.

Reply:
+1 to what @gazarenkov said. But I think that there is a confusion:
- @gazarenkov right now username is used in this field, and @davidfestal needs the user ID
- @davidfestal my vote is for setting the user ID in the owner field, neither renaming nor adding any field. I believe it is what @sleshchenko and @gazarenkov was talking about

Reply:
Until now the Keycloak `preferred_name` (=> user name such as `dfestal` => login name afaik) was used in `RuntimeIdentity`, instead of the real `userId` that is the real user *primary key* afaik. In this PR I add the `userId` in the `RuntimeIdentity` because:
1. We need the `userId` in OSIO
2. I think it has its natural place in the `RuntimeIdentity` object (as you just explained)

However, since I don't fully understand why the user name was used previously, and what precise requirements is used to meet, I kept it in the `RuntimeIdentity` object for now: it has the advantage of making it directly available in every place where it is used, instead of having to retrieve it from the user ID.

My proposal is to merge this PR with the 2 fields, since it doesn't hurt and makes sure there's no regression, and then study the removal of the `userName` part in a distinct Pull Request.

Reply:
@garagatyi I'm OK with this, but I still think it would be better to do this is a distinct PR.

Reply:
-1 to merge with an additional field. I think we should avoid extending code in different places and rework it in the following PRs. The best code is not written code. And using userId instead of username in owner field would remove a lot of code from this PR and keep things simple. 
About the reasoning for initial putting username into the owner field, I believe that it was done unintentionally. AFAIK Che 6 is not GA yet, so backward compatibility should not be an issue in this case.

Reply:
We just need to fill owner with unique user identifier (whatever it is called in user identity system: id, name, key, etc). Since this class is not intended to be persisted (in normal case), I see no risk changing it, especially before GA.

Reply:
Iterating with 2 distinct PR should not to be considered bad by itself. I prefer multiple small PRs instead of one-month-long PR that is more likely to break something, more difficult to test and review.

I think we all agree that `ownerName` should be replace by `ownerID` but if David wants to split that in 2 steps (1. add `userId` 2. remove `userName`) that make sense to me.

The value of this PR is not the refactoring of `RuntimeIdentity` but to make Che work in multi tenant scenario like OSIO.

@gorkem I'm afraid that's one of the cases where we need your help to take a decision without spending too much time discussing.

Reply:
@l0rd as far as I know the idea is to avoid adding changes to the class rather than changing it twice.
What is suggested is to change the content of the field which should be 1 line change, and remove dozens of lines of changes.

Reply:
@l0rd it is definitely not the case you have in mind, so procedure you propose is hardly applicable. I do not even think it looks like a situation (yet at least) where code maintainer should be involved (as it stated here https://github.com/eclipse/che/wiki/Development-Process ). Let engineering discussion go and the best/simplest way win :)

Reply:
@garagatyi yeah but [good refactoring practices](https://www.amazon.com/Refactoring-Improving-Design-Existing-Code/dp/0201485672) (an excellent book by the way) suggest not to take shortcuts. When you need to replace some code the advice is to split it in 2 simpler steps (addition of new code and removal of existing code).

@gazarenkov unfortunately it looks like there is no consensus. This is an urgent PR: it allows to deploy Che 6 on OSIO. Given the importance I would like to move forward as soon as possible. I don't want this PR to be subject to a multiple weeks review process because of a trivial refactoring when the problem it solves has been afflicting us for months.

Reply:
backing to the discussion in substance.
As I understand all agreed that we need only one field (owner) which currently contains user name but it is better if it contain userID.

As I understand the multistep procedure ((1) add OwnerID; (2) remove Owner(Name)) proposed due to potential backward compatibility risk.

My questions are:
What is the backward compatibility risk you, guys, (@davidfestal @l0rd @skabashnyuk) see if we just filling Owner with userID (one line code change as @garagatyi mentioned) taking into account that: this class's content is not exposed externally (not used in API etc)?

@davidfestal just for my understanding,  when you said "We need the userId in OSIO" what exactly purpose/case you have in mind in this regard?

thanks

Reply:
@gazarenkov the fact that `owner` was not used in API doesn't mean that it's not used at all and that we can remove it and be sure that there will be no regression. That's just one more modification that need attention and testing and @davidfestal prefer to do it in a separate PR. And that's absolutely legitimate taken into account the complexity of the actual PR.

Your second question (why we need the userID) is actually answered at point 3 of this PR description.

I am sure that there are still tons of questions that can be asked on this PR and that there are a lot of refactoring that could still be done (is code ever perfect?) but considered this PR importance and that @davidfestal has answered and addressed almost all comments I think that we should merge and move forward. @sleshchenko @garagatyi wdyt?

Reply:
@l0rd thanks for answering second question

still confusing for first...
Should I read it as: there are no visible risk but you prefer not to change the current PR (due to complexity etc). Which means once merged this PR will be immediately followed by another which removes userName?
If it is the case - I let you, guys, decide on tactic.
Otherwise - I still have conceptual concerns I voiced initially and ask for describing the risk you can see in details.

Reply:
@gazarenkov yes this PR will be followed by another one which removes `userName`

Reply:
I created issue #8859 as a placeholder for further discussion on the userName - userId replacement.